### PR TITLE
New course enrichment

### PR DESCRIPTION
**Pull request by defong**
### Context
Covers 

- 'Course summary page' & word capping
- 'About this course' enrichment and saving & validation
- 'Requirements and eligibility' enrichment and saving & validation

### Changes proposed in this pull request

Allowing the user to enrich a course and providing a course summary of the enrichments

### Guidance to review

'Course summary page' & word capping
empty-fields

at-100

'About this course' enrichment and saving & validation
about-this-course
about-this-course-validation

'Requirements and eligibility' enrichment and saving & validation
requirements
requirements-validation

### CODE REVIEW

**Comment by defong** [comment]
['Requirements and eligibility' enrichment and saving](https://trello.com/c/NT5LrgjT/172-requirements-and-eligibility-enrichment-and-saving)

**Comment by defong** [comment]
['About this course' enrichment and saving](https://trello.com/c/e1Q2GDfM/171-about-this-course-enrichment-and-saving)

**Comment by defong** [comment]
['About this course' enrichment and saving](https://trello.com/c/e1Q2GDfM/171-about-this-course-enrichment-and-saving)

**Comment by fofr** [comment]
For the truncated text on the summary, @dankmitchell can we use some CSS like this as an enhancement, to keep it to one line: https://github.com/fofr/manage-courses-prototype/blob/master/app/assets/sass/patterns/_course-parts.scss#L115-L118

**Comment by fofr** [comment]
The titles on each page need to have a secondary heading showing which course it is, like this:
screen shot 2018-08-24 at 12 02 01

**Comment by fofr** [comment]
Can we put the formatting text we used on about your org in the right hand column too?

screen shot 2018-08-24 at 12 03 35

**Comment by fofr** [comment]
The spacing beneath the textareas looks a little large @dankmitchell?

File: src/ui/ViewModels/FromUcasViewModel.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@ public class FromUcasViewModel
         public string CourseTitle { get; set; }
         public string AccreditingProviderId { get; set; }
         public CourseVariantViewModel Course { get; set; }
+        public CourseEnrichmentViewModel CourseEnrichment { get; set; }
```
Comment (timabell) [inline src/ui/ViewModels/FromUcasViewModel.cs]: Why is there enrichment in a FromUcas model?
Does the model need renaming?

Minor.

File: ManageCoursesUi.Tests/CourseControllerTests.cs [hunk 2]
```diff
@@ -167,7 +191,222 @@ public void TestController_with_api_exception_on_second_call_should_return_excep
...
+
+            var controller = new CourseController(manageApi.Object);
+            var result = await controller.About(TestHelper.InstitutionCode, TestHelper.AccreditedProviderId, TestHelper.TargetedUcasCode);
+
+            var viewResult = result as ViewResult;
+            Assert.IsNotNull(viewResult);
+
+            var model = viewResult.Model as AboutCourseEnrichmentViewModel;
+
+            Assert.IsNotNull(model);
```
Comment (timabell) [inline ManageCoursesUi.Tests/CourseControllerTests.cs]: I'd like us to use fluent assertions everywhere.

Reject

File: src/ui/ViewModels/FromUcasViewModel.cs (hunk 1 above)
Comment (defong) [reply src/ui/ViewModels/FromUcasViewModel.cs]: Yeah the FromUcas needs a rename, reason why the enrichment is there is simply cause FromUcas is the varient page view model

**Pull request merged**